lib/linter/apply-disable-directives.js

```diff
@@ -228,6 +230,14 @@ function applyDirectives(options) {
                     } else {
                         disabledRuleMap.set(directive.ruleId, directive);
                     }
+
+                    if (directive.ruleId === null || directive.ruleId === problem.ruleId) {
```

Comment:
In what situation would ruleId be null?

Reply:
When we don't specify the rules to disable. For example:
```js
/* eslint-disable */
alert("test");
```
In this case `ruleId` is null.

The ruleId of internal messages/problems (e.g., unused eslint-disable directive) is also null. But these are not affected.

Reply:
Ah, found a bug related to `--report-unused-disable-directives` in my implementation. Will fix asap.